### DISCUSSION TITLE
[cloud_firestore] implementation of `FieldPath.of()`

### DISCUSSION THREAD

**Discussion by Reprevise** [firebase/flutterfire#6612] [Feature request]
## Bug report

**Describe the bug**
I'm trying to update a field that has special characters in its name. The error message says to use `FieldPath.of()` which does not exist.

Error message:
> Use FieldPath.of() for field names containing '~*/[]'.)

I've referenced [this answer](https://stackoverflow.com/a/62658347/10808802) from a Firebase engineer who says it's currently not possible to query/update a field that contains special characters since there's no wrapper for `FieldPath.of()`

### Steps to reproduce

1. Create/update a field that contains special characters (~*/[])
2. See error or incorrect behavior

### Expected behavior

The field should be created/updated.

**Comment by alextat120** [comment]
Can we also support FieldPath in the update method like in JS?

https://github.com/FirebaseExtended/flutterfire/discussions/5057
